src/actions/app.js

```diff
@@ -0,0 +1,28 @@
+import { push } from 'react-router-redux';
+import DevTools from '../service/dev-tools.js';
+import constants from '../constants.js';
+
+const devTools = new DevTools();
+
+export const INIT = 'INIT';
+export const TOGGLE_DEVTOOLS = 'TOGGLE_DEVTOOLS';
+
+export function init(data) {
+  return (dispatch) => {
+    dispatch({
+      type: INIT,
+      ...data
+    });
+    dispatch(push('/'));
+
+    if (constants.DEV) {
+      dispatch(toggleDevTools());
```

Comment:
There's a few reasons:
- `dispatch(foo())` is a redux-ism, where `foo()` is an _action creator_ that returns a _flux standard action_ (basically, an object with the property `type`). So, the `toggleDevTools()` in your comment wouldn't make sense with `dispatch`, as dispatch expects an action to pass to it's reducers.
- `init` itself is an action creator with a thunk. Normally an action creator returns a single action. This one returns a function, so I can group a number of actions from one action.
- The `toggleDevTools` action is also already used in other places (keyboard shortcuts). Making an action for it, rather than what you suggested allows other parts of the app to hook into the action. I'm keeping extensibility in mind for down the road for plugins (hyperterm is an epic example of this).